front/app/containers/Admin/reporting/context/ReportContext.tsx

```diff
@@ -1,3 +1,58 @@
-import { createContext } from 'react';
+import React, { createContext, useContext } from 'react';
+import useReport from 'api/reports/useReport';
+import usePhase from 'api/phases/usePhase';
 
-export const ReportContext = createContext<'phase' | 'pdf'>('pdf');
+interface ReportContextBase {
+  width: 'pdf' | 'responsive';
+  reportId: string;
+}
+
+export interface ReportContextProps extends ReportContextBase {
+  children: React.ReactNode;
+}
+
+interface ReportContextData extends ReportContextBase {
+  projectId?: string;
+  phaseId?: string;
+}
+
+// In practice, the context will never have these default values- just put them here to please TS
+const Context = createContext<ReportContextData>({
+  width: 'pdf',
+  reportId: '',
+});
+
+export const ReportContextProvider = ({
```

Comment:
Expanded this context a bit. Makes it easy to have access to the report that you're in, and the corresponding phase and project id (if any). Saves a lot of prop drilling and hook calling